app/scripts/models/reliers/oauth.js

```diff
@@ -16,9 +16,49 @@ define(function (require, exports, module) {
   var RelierKeys = require('lib/relier-keys');
   var Url = require('lib/url');
   var Validate = require('lib/validate');
+  var Validator = require('vat');
 
   var RELIER_FIELDS_IN_RESUME_TOKEN = ['state', 'verificationRedirect'];
 
+  /*eslint-disable camelcase*/
+  var CLIENT_INFO_SCHEMA = {
+    id: Validator.string().required().use(Validate.isHexValid).as('clientId'),
+    image_uri: Validator.string().optional().allow('').use(Validate.isUrlValid).as('imageUri'),
```

Comment:
> joi marks items optional by default, think we should do the same to reduce boilerplate?

Sounds good. You could also make it possible to configure the default on initialisation? That way it could be `optional` for people who want that behaviour and `required` for people who want the opposite.